core/src/apps/common/sdcard.py

```diff
@@ -13,61 +11,105 @@ class SdCardUnavailable(wire.ProcessError):
+            verb="Retry",
+            verb_cancel="Abort",
+            icon=ui.ICON_WRONG,
+            larger_vspace=True,
+            exc=SdCardUnavailable("SD card required."),
+        )
     else:
-        text.normal("Please unplug the", "device and insert your", "SD card.")
-        btn_confirm = None
-        btn_cancel = "Close"
-
-    return await confirm(ctx, text, confirm=btn_confirm, cancel=btn_cancel)
+        await show_error_and_raise(
+            ctx,
+            "warning_no_sd",
+            header="SD card protection",
+            subheader="SD card required.",
+            content="Please unplug the\ndevice and insert your SD card.",
+            exc=SdCardUnavailable("SD card required."),
+        )
 
 
-async def format_card_dialog(ctx: wire.GenericContext) -> bool:
+async def _confirm_format_card(ctx: wire.GenericContext) -> None:
     # Format card? yes/no
-    text = Text("SD card error", ui.ICON_WRONG, ui.RED)
-    text.bold("Unknown filesystem.")
-    text.br_half()
-    text.normal("Use a different card or")
-    text.normal("format the SD card to")
-    text.normal("the FAT32 filesystem.")
-    if not await confirm(ctx, text, confirm="Format", cancel="Cancel"):
-        return False
+    await confirm_action(
+        ctx,
+        "warning_format_sd",
+        "SD card error",
+        action="Unknown filesystem.",
+        description="Use a different card or format the SD card to the FAT32 filesystem.",
+        icon=ui.ICON_WRONG,
+        icon_color=ui.RED,
+        verb="Format",
+        verb_cancel="Cancel",
+        larger_vspace=True,
+        exc=SdCardUnavailable("SD card not formatted."),
+    )
 
     # Confirm formatting
-    text = Text("Format SD card", ui.ICON_WIPE, ui.RED)
-    text.normal("Do you really want to", "format the SD card?")
-    text.br_half()
-    text.bold("All data on the SD card", "will be lost.")
-    return await hold_to_confirm(ctx, text, confirm="Format SD card")
-
-
-async def sd_problem_dialog(ctx: wire.GenericContext) -> bool:
-    text = Text("SD card problem", ui.ICON_WRONG, ui.RED)
-    text.normal("There was a problem", "accessing the SD card.")
-    return await confirm(ctx, text, confirm="Retry", cancel="Abort")
+    await confirm_action(
+        ctx,
+        "confirm_format_sd",
+        "Format SD card",
+        action="All data on the SD card will be lost.",
+        description="Do you really want to format the SD card?",
+        reverse=True,
+        verb="Format SD card",
+        icon=ui.ICON_WIPE,
+        icon_color=ui.RED,
+        hold=True,
+        larger_vspace=True,
+        exc=SdCardUnavailable("SD card not formatted."),
+    )
+
+
+async def confirm_retry_sd(
+    ctx: wire.GenericContext,
+    exc: wire.ProcessError = SdCardUnavailable("Error accessing SD card."),
+) -> None:
+    await confirm_action(
+        ctx,
+        "warning_sd_retry",
+        "SD card problem",
+        action="None",
```

Comment:
didn't notice it before but this should probably be `None`, not `"None"`

Reply:
also means this is not covered by UI tests, correct?
that's probably because the SD card persistence tests don't have UI recording...

Reply:
Oof, fixed in 82b690518f6c60f4d81c22421c45ad3e06322ee1. Will look into writing a testcase.

Reply:
There doesn't seem to be a simple way to inject SD card errors that lead to these dialogs:/ Checked manually that both work and look OK.